model/load_balancer.rb

```diff
@@ -43,6 +43,12 @@ def add_vm(vm)
     end
   end
 
+  def detach_vm(vm)
+    load_balancers_vms_dataset.where(vm_id: vm.id).update(state: "detaching")
```

Comment:
nit: it may have same syntax with `remove_vm`
`load_balancers_vms_dataset[vm_id: vm.id].update(state: "detaching")`